src/hooks/useFormData.ts

```diff
@@ -0,0 +1,71 @@
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { ERROR_MESSAGES } from '../messages';
+
+export type FetchedAddress = {
+  prefecture: string;
+  area: string;
+};
+
+export type AddressForm = FetchedAddress & { postalCode: string };
+
+export const useFormData = () => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    formState: { errors },
+  } = useForm<AddressForm>({
+    mode: 'onBlur',
+    defaultValues: {
+      postalCode: '',
+      prefecture: '',
+      area: '',
+    },
+  });
+
+  const [noResultsMessage, setNoResultsMessage] = useState<string>('');
+
+  const onSubmit: SubmitHandler<AddressForm> = (data) => console.log(data);
+  const fetchData = async (query: string) => {
+    try {
+      const response = await fetch(`https://zipcloud.ibsnet.co.jp/api/search?zipcode=${query}`);
+      const result = await response.json();
+      if (result['results']) {
```

Comment:
This condition will probably be true even when `result['results']` is an empty object. I think you may need to check for lengths too.

```tsx
      if (result.results?.length) {
```

But instead of this, I suggest you do early returns instead because I think it's much more readable.

```tsx
const result = await response.json();
if (!result?.results?.length) return null // return null instead of constructing empty format 

 const { address1: prefecture, address2: area } = result.results[0];
return { prefecture, area };
```

This is the same as doing
```if (!result || !result.results || result.results.length === 0) ``` 

`?` is the optional chaining that will just return `undefined` instead of throwing an error. Read more about this here: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

Reply:
Added length check for results